### PR TITLE
make it possible to have one color per glyph

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

File: src/layouting/layouting.jl [hunk 1]
```diff
@@ -216,10 +228,10 @@ function glyph_collection(str::AbstractString, font_per_char, fontscale_px, hali
         reduce(vcat, charorigins),
         [x.extent for x in charinfos],
         [Vec2f(x.scale) for x in charinfos],
-        [rotation for x in charinfos],
-        [color for x in charinfos],
-        [strokecolor for x in charinfos],
-        [strokewidth for x in charinfos],
+        per_char_info(rotation, charinfos),
```
Comment (jkrumbiegel) [inline src/layouting/layouting.jl:231]: hm neither what I had before nor this seems ideal. there is already a function called `attribute_per_char` at the top, this is currently used for font and textsize, but similar logic could apply to color. anyway, it feels weird to do one expansion before the glyph_collection function and one here.

**Review by jkrumbiegel — COMMENTED** [review]
this is one of these places in the code base where it's difficult to know what type `color` etc could have, and therefore how the code will execute. also, I guess we recompile this expensive looking function for a color given as a Symbol vs. as an RGBf etc. I would like clearer logic, or does that tie into the attribute refactor idea?

**Comment by SimonDanisch** [comment]
>I would like clearer logic, or does that tie into the attribute refactor idea?

To separate the concerns a bit, I would leave this for the attribute refactor PR, as long as this PR doesn't introduce any regressions

File: src/layouting/layouting.jl (hunk 1 above)
Comment (SimonDanisch) [reply src/layouting/layouting.jl:231]: Ah, I somehow didnt find that function... So I guess I should just use that?

Comment (SimonDanisch) [reply src/layouting/layouting.jl]: Ok, this uses `attribute_per_char` now!

**Pull request merged**